src/core/unit.h

```diff
@@ -464,6 +464,9 @@ typedef struct Unit {
         /* When writing transient unit files, stores which section we stored last. If < 0, we didn't write any yet. If
          * == 0 we are in the [Unit] section, if > 0 we are in the unit type-specific section. */
         signed int last_section_private:2;
+
+        /* Remember this unit was part of some dependency cycle. */
+        bool was_on_dependency_cycle;
```

Comment:
We discussed this with @msekletar and he described an idea where the units that were dropped because of a cycle would not  be garbage collected. This seems reasonable, considering that it's just another type of failure, and we do not gc failed units.

Reply:
I still think the dep cycle is more of a property of Job rather than Unit, and should be reported as such. It should not be recorded forever in unit state...

Reply:
You could say that same about any other failure… "It's a property of a Job", e.g. because the unit was started too early. And it's not recorded "forever": just until a successful start or reset-failures.